libs/angular-remote-components/src/lib/components/slot/slot.component.ts

```diff
@@ -43,23 +40,38 @@ export class SlotComponent implements OnInit, OnDestroy {
         if (viewContainers && viewContainers.length === components.length) {
           components.forEach((componentInfo, i) => {
             if (componentInfo.componentType) {
-              const componentRef = viewContainers.get(i)?.createComponent<any>(componentInfo.componentType)
-              if (componentRef && 'ocxInitRemoteComponent' in componentRef.instance) {
-                ;(componentRef.instance as ocxRemoteComponent).ocxInitRemoteComponent({
-                  appId: componentInfo.remoteComponent.appId,
-                  productName: componentInfo.remoteComponent.productName,
-                  baseUrl: componentInfo.remoteComponent.baseUrl,
-                  permissions: componentInfo.permissions,
-                })
-              }
-              componentRef?.changeDetectorRef.detectChanges()
+              Promise.all([Promise.resolve(componentInfo.componentType), Promise.resolve(componentInfo.permissions)]).then(([componentType, permissions]) => {
+                this.createComponent(componentType, componentInfo, permissions, viewContainers, i)
```

Comment:
Set loading property to false after createComponent

You may have to manually trigger the change detection after that for the change to be taken up